internal/host/host.go

```diff
@@ -321,6 +322,24 @@ func (m *Manager) updateInventory(ctx context.Context, cluster *common.Cluster,
 		return common.NewApiError(http.StatusNotFound, err)
 	}
 
+	if m.Config.BootstrapHostMAC != "" && !h.Bootstrap {
+		for _, iface := range inventory.Interfaces {
+			if iface.MacAddress == m.Config.BootstrapHostMAC {
+				log.Infof("selected local bootstrap host %s for cluster %s", h.ID, cluster.ID)
+				err = m.UpdateRole(ctx, h, models.HostRoleMaster, db)
```

Comment:
nit: This line kinda makes [this](https://github.com/openshift/assisted-service/blob/0cb3eb4601060c9a4111a3bc99da3c9cc086d115/internal/host/host.go#L615-L616) code comment slightly incorrect, might want to adjust it

Reply:
What would be a correct version of that comment?

Reply:
It says it's always invoked from an API (referring to the host role update api) which is kinda untrue because now it's invoked internally from the inventory API. Also it says that the role is user selected but it's not, it's selected automatically in this case with the mac mechanism.

I'm not sure exactly what's the point of that comment, so I'm not sure what would be the correct phrasing, but I can tell after this change it's at least inaccurate, so something should be done. Either delete it or figure out its purpose and rephrase it somehow

Reply:
I replaced the call to `m.UpdateRole()` with a direct call to `updateRole()` so that none of us need to decide what the comment means :smile: